backend/src/contracts/tests.rs

```diff
@@ -1,24 +1,69 @@
-mod test {
+use std::{sync::Arc, time::Duration};
+
+use ethers::{
+    prelude::{ContractFactory, SignerMiddleware},
+    providers::{Http, Provider},
+    signers::{LocalWallet, Signer},
+    types::{H160, U256},
+};
+
+use crate::contracts::generated::mock_erc20::{MockERC20, MOCKERC20_ABI, MOCKERC20_BYTECODE};
+
+pub async fn initialize_anvil(
+    anvil: &ethers::utils::AnvilInstance,
+) -> (
+    H160,
+    H160,
+    Arc<SignerMiddleware<Provider<Http>, LocalWallet>>,
+    MockERC20<SignerMiddleware<Provider<Http>, LocalWallet>>,
+) {
+    let cex_addr_1 = anvil.addresses()[1];
+    let cex_addr_2 = anvil.addresses()[2];
+    let wallet: LocalWallet = anvil.keys()[0].clone().into();
+    let provider = Provider::<Http>::try_from(anvil.endpoint())
+        .unwrap()
+        .interval(Duration::from_millis(10u64));
+
+    let client = Arc::new(SignerMiddleware::new(
+        provider,
+        wallet.with_chain_id(anvil.chain_id()),
+    ));
+
+    let factory = ContractFactory::new(
+        MOCKERC20_ABI.to_owned(),
+        MOCKERC20_BYTECODE.to_owned(),
+        Arc::clone(&client),
+    );
+
+    let mock_erc20_deployment = factory.deploy(()).unwrap().send().await.unwrap();
+
+    let mock_erc20 = MockERC20::new(mock_erc20_deployment.address(), Arc::clone(&client));
+
+    let mint_call = mock_erc20.mint(cex_addr_2, U256::from(556863));
+    assert!(mint_call.send().await.is_ok());
+
+    return (cex_addr_1, cex_addr_2, client, mock_erc20);
+}
 
-    use std::{time::Duration, sync::Arc};
+mod test {
+    use std::sync::Arc;
 
     use ethers::{
-        prelude::{ContractFactory, SignerMiddleware},
-        providers::{Http, Middleware, Provider, StreamExt},
-        signers::{LocalWallet, Signer},
-        types::{U256, Filter, Address},
-        utils::{Anvil},
+        types::{Address, Filter, U256},
+        utils::Anvil,
     };
 
-    use crate::{
-        contracts::{generated::{
-            mock_erc20::{self, MockERC20, MOCKERC20_ABI, MOCKERC20_BYTECODE},
-            summa_contract::{ProofOfSolvencySubmittedFilter, Summa, ExchangeAddressesSubmittedFilter},
+    use crate::contracts::{
+        generated::{
+            summa_contract::{
+                ExchangeAddressesSubmittedFilter, ProofOfSolvencySubmittedFilter, Summa,
+            },
             verifier::SolvencyVerifier,
-        }, signer::SummaSigner},
+        },
+        signer::SummaSigner,
+        tests::initialize_anvil,
     };
 
-
     #[tokio::test]
     async fn test_sign_message() {
         let anvil = Anvil::new().spawn();
```

Comment:
I think this can also go into the initialization function since it's used in multiple places and the mnemonic is the same.

Reply:
Create `AnvilInstance` in the `initialize_anvil` method 8502144c